pkg/gameserverallocations/allocator.go

```diff
@@ -567,6 +570,34 @@ func (c *Allocator) allocationUpdateWorkers(ctx context.Context, workerCount int
 	return updateQueue
 }
 
+// applyAllocationToGameServer patches the inputted GameServer with the allocation metadata changes, and updates it to the Allocated State.
+// Returns the updated GameServer.
+func (c *Allocator) applyAllocationToGameServer(ctx context.Context, fam allocationv1.MetaPatch, gs *agonesv1.GameServer) (*agonesv1.GameServer, error) {
+	// patch ObjectMeta labels
+	if fam.Labels != nil {
+		if gs.ObjectMeta.Labels == nil {
+			gs.ObjectMeta.Labels = make(map[string]string, len(fam.Labels))
+		}
+		for key, value := range fam.Labels {
+			gs.ObjectMeta.Labels[key] = value
+		}
+	}
+
+	if gs.ObjectMeta.Annotations == nil {
+		gs.ObjectMeta.Annotations = make(map[string]string, len(fam.Annotations))
+	}
+	// apply annotations patch
+	for key, value := range fam.Annotations {
+		gs.ObjectMeta.Annotations[key] = value
+	}
+
+	// add last allocated, so it always gets updated, even if it is already Allocated
+	gs.ObjectMeta.Annotations["agones.dev/last-allocated"] = time.Now().String()
```

Comment:
Format of string is:  "2006-01-02 15:04:05.999999999 -0700 MST", which includes the timezone, so I think we should be good there for readability.

For the sake of the purposes we need it for allocation, we actually only care that the value changes, so that the object gets updated within K8s, without a field that has changed, if we update the `GameServer` with no differences from what is stored in etcd, it becomes a noop, which means it can't be watched for by the SDK or other services.